frontend/src/pages/Account.tsx

```diff
@@ -0,0 +1,280 @@
+import {useEffect, useState} from "react";
+import {
+    AlertColor,
+    Box,
+    Container, Typography,
+} from "@mui/material";
+import Button from '@mui/material/Button';
+import Grid from '@mui/material/Unstable_Grid2';
+import TextField from '@mui/material/TextField';
+import Autocomplete from '@mui/material/Autocomplete';
+import {DateField} from "@mui/x-date-pickers";
+import Chip from '@mui/material/Chip';
+import Paper from '@mui/material/Paper';
+import {styled} from '@mui/material/styles';
+import Alert from '@mui/material/Alert';
+import IconButton from '@mui/material/IconButton';
+import CloseIcon from '@mui/icons-material/Close';
+import CircularProgress from '@mui/material/CircularProgress';
+import Stack from '@mui/material/Stack';
+import {AdapterDayjs} from '@mui/x-date-pickers/AdapterDayjs';
+import {LocalizationProvider} from '@mui/x-date-pickers/LocalizationProvider';
+import dayjs from "dayjs";
+
+import {API_BASE_URL, signIn, signOut} from '../components/Api';
+import {Province, ChipData, chipColor, User} from "../components/Types";
+import httpClient from "../httpClient";
+import {GetProvincesWithNA} from "../components/Provinces";
+
+export const Account = () => {
+    const {t} = useTranslation();
+
+    const [province, setProvince] = useState<Province | null>(null);
+    const [chipData, setChipData] = useState<readonly ChipData[]>([]);
+
+    const [alert, setAlert] = useState(false);
+    const [alertType, setAlertType] = useState<AlertColor>("error");
+    const [alertContent, setAlertContent] = useState('');
+
+    const provinces: Province[] = GetProvincesWithNA();
+
+    const [user, setUser] = useState<User | null>(null);
+    const [loading, setLoading] = useState(true);
+
+    useEffect(() => {
+        (async () => {
+            try {
+                const resp = await httpClient.get(API_BASE_URL + "/user_info");
+                setUser(resp.data);
+            } catch (error) {
+                console.log("Not authenticated");
+            }
+            setLoading(false);
+        })();
+    }, []);
+
+    let default_province: Province = {label: 'N/A', id: 'na', region: 'N/A', region_id: 'na'};
+    let default_dob = dayjs('2022-01-01');
+    let default_WCAID = "";
+    if (user != null) {
+        if (user.province != null) {
+            //set province in the combo box
+            for (let i = 0; i < provinces.length; i++) {
+                if (provinces[i].id === user.province) {
+                    default_province = provinces[i];
+                }
+            }
+        }
+        if (user.dob != null) {
+            default_dob = dayjs(user.dob);
+        }
+        if (user.roles != null && user.roles.length > 0 && chipData.length === 0) {
+            let tmpChipData = [];
+            for (let i = 0; i < user.roles.length; i++) {
+                tmpChipData.push({key: i, label: user.roles[i]});
+            }
+            setChipData(tmpChipData);
```

Comment:
This is an effect, so it should be in a `useEffect` - or ideally, just calculate the initial value directly in the chip data `useState` above:
```
const [chipState, setChipState] = useState<readonly ChipData[]>(
  // function to calculate initial value here
);
```

Reply:
well it's not really doable (to calculate the initial value directly I mean) I think because we get the value once the person is logged in the, so all the data really just depends on pinging the API

Reply:
Oh, right.
Then lines 102 to 110 should be wrapped in a `useEffect` with dependency array `[user]` (and you won't need the `chipData.length === 0` check I think).